source/Calamari.Tests/KubernetesFixtures/Helm/HelmTemplateValueSourcesCreatorFixture.cs

```diff
@@ -0,0 +1,248 @@
+            };
+            var fileSystem = Substitute.For<ICalamariFileSystem>();
+
+            // Act
+            Action act = () => HelmTemplateValueSourcesParser.ParseTemplateValuesSources(deployment, fileSystem, new SilentLog());
+
+            // Assert
+            act.Should().ThrowExactly<ArgumentException>();
+        }
+        
+        [Test]
+        public void ParseTemplateValuesSources_ChartSourceButNotScriptSourceVairable_ShouldThrowArgumentNullException()
+        {
+            // Arrange
+            var templateValuesSourcesJson = JsonConvert.SerializeObject(new HelmTemplateValueSourcesParser.TemplateValuesSource[]
+                                                                        {
+                                                                            new HelmTemplateValueSourcesParser.ChartTemplateValuesSource
+                                                                            {
+                                                                                ValuesFilePaths = "secondary.yaml"
+                                                                            }
+                                                                        },
+                                                                        Formatting.None);
+
+            var variables = new CalamariVariables
+            {
+                [SpecialVariables.Helm.TemplateValuesSources] = templateValuesSourcesJson,
+                [KnownVariables.OriginalPackageDirectoryPath] = RootDir,
+                [PackageVariables.IndexedPackageId(string.Empty)] = "mychart",
+                [PackageVariables.IndexedPackageVersion(string.Empty)] = "0.3.8"
+            };
+
+            var deployment = new RunningDeployment(variables)
+            {
+                CurrentDirectoryProvider = DeploymentWorkingDirectory.StagingDirectory
+            };
+            var fileSystem = Substitute.For<ICalamariFileSystem>();
+
+            // Act
+            Action act = () => HelmTemplateValueSourcesParser.ParseTemplateValuesSources(deployment, fileSystem, new SilentLog());
+
+            // Assert
+            act.Should().ThrowExactly<ArgumentNullException>();
+        }
+
+        [Test]
+        public void ParseTemplateValuesSources_ChartPackageAndInlineAndKeyValues_CorrectlyParsesAndWritesAndOrdersFiles()
+        {
+            // Arrange
+            var templateValuesSourcesJson = JsonConvert.SerializeObject(new HelmTemplateValueSourcesParser.TemplateValuesSource[]
+                                                                        {
+                                                                            new HelmTemplateValueSourcesParser.InlineYamlTemplateValuesSource
+                                                                            {
+                                                                                Value = @"it:
+  is:
+    some: 'yaml'"
```

Comment:
Not sure if we can actually do it in Calamari due to c# versions, but a raw string literal would be perfect here.

Reply:
Yeah, we can't unfortunately due to `.Net 4.6.2`